sdks/java/core/src/test/java/org/apache/beam/sdk/io/TFRecordIOTest.java

```diff
@@ -209,15 +208,13 @@ public void testWriteTwo() throws Exception {
   @Test
   @Category(NeedsRunner.class)
   public void testReadInvalidRecord() throws Exception {
-    expectedException.expect(IllegalStateException.class);
```

Comment:
(here and below) This was previously throwing the top level exception but DirectRunner wraps ParDo exceptions with PipelineExecutionException but BoundedSource/UnboundedSource used a different code path and hence a different exception handling path.